guidedmodules/views.py

```diff
@@ -1023,22 +1024,43 @@ def authoring_edit_module(request, task):
 
 
 @login_required
-def change_task_state(request):
+@transaction.atomic
+def delete_task(request):
+    # Mark a Task as deleted and un-link it from any questions it
+    # answers by saving new answers with the Task removed. Only
+    # Project admins can delete tasks. It's not a destructive
+    # operation at the database level, but it does cause the Task
+    # to become inaccessible.
+
     if request.method != "POST":
         return HttpResponseNotAllowed(["POST"])
 
     task = get_object_or_404(Task, id=request.POST["id"], project__organization=request.organization)
-    if not task.has_write_priv(request.user, allow_access_to_deleted=True):
+    if not request.user in task.project.get_admins():
```

Comment:
It seems that has_write_priv is mor granular than get_admins. Can you elaborate?

Reply:
Thanks.